File: src/main/java-graphql/io/github/graphql/model/ContributionCalendarMonthResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for ContributionCalendarMonth
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class ContributionCalendarMonthResponseProjection extends GraphQLResponseProjection {

    public ContributionCalendarMonthResponseProjection() {
    }

    @Override
    public ContributionCalendarMonthResponseProjection all$() {
        return all$(3);
    }

    @Override
    public ContributionCalendarMonthResponseProjection all$(int maxDepth) {
        this.firstDay();
        this.name();
        this.totalWeeks();
        this.year();
        this.typename();
        return this;
    }

    public ContributionCalendarMonthResponseProjection firstDay() {
        return firstDay(null);
    }

    public ContributionCalendarMonthResponseProjection firstDay(String alias) {
        fields.add(new GraphQLResponseField("firstDay").alias(alias));
        return this;
    }

    public ContributionCalendarMonthResponseProjection name() {
        return name(null);
    }

    public ContributionCalendarMonthResponseProjection name(String alias) {
        fields.add(new GraphQLResponseField("name").alias(alias));
        return this;
    }

    public ContributionCalendarMonthResponseProjection totalWeeks() {
        return totalWeeks(null);
    }

    public ContributionCalendarMonthResponseProjection totalWeeks(String alias) {
        fields.add(new GraphQLResponseField("totalWeeks").alias(alias));
        return this;
    }

    public ContributionCalendarMonthResponseProjection year() {
        return year(null);
    }

    public ContributionCalendarMonthResponseProjection year(String alias) {
        fields.add(new GraphQLResponseField("year").alias(alias));
        return this;
    }

    public ContributionCalendarMonthResponseProjection typename() {
        return typename(null);
    }

    public ContributionCalendarMonthResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}

File: src/main/kotlin-graphql/io/github/graphql/kotlin/model/ContributionCalendarMonthResponseProjection.kt
package io.github.graphql.kotlin.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection
import java.util.Objects

/**
 * Response projection for ContributionCalendarMonth
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-07T19:48:13+0800"
)
open class ContributionCalendarMonthResponseProjection : GraphQLResponseProjection() {

    override fun `all$`(): ContributionCalendarMonthResponseProjection = `all$`(3)

    override fun `all$`(maxDepth: Int): ContributionCalendarMonthResponseProjection {
        this.firstDay()
        this.name()
        this.totalWeeks()
        this.year()
        this.typename()
        return this
    }

    fun firstDay(): ContributionCalendarMonthResponseProjection = firstDay(null)

    fun firstDay(alias: String?): ContributionCalendarMonthResponseProjection {
        fields.add(GraphQLResponseField("firstDay").alias(alias))
        return this
    }

    fun name(): ContributionCalendarMonthResponseProjection = name(null)

    fun name(alias: String?): ContributionCalendarMonthResponseProjection {
        fields.add(GraphQLResponseField("name").alias(alias))
        return this
    }

    fun totalWeeks(): ContributionCalendarMonthResponseProjection = totalWeeks(null)

    fun totalWeeks(alias: String?): ContributionCalendarMonthResponseProjection {
        fields.add(GraphQLResponseField("totalWeeks").alias(alias))
        return this
    }

    fun year(): ContributionCalendarMonthResponseProjection = year(null)

    fun year(alias: String?): ContributionCalendarMonthResponseProjection {
        fields.add(GraphQLResponseField("year").alias(alias))
        return this
    }

    fun typename(): ContributionCalendarMonthResponseProjection = typename(null)

    fun typename(alias: String?): ContributionCalendarMonthResponseProjection {
        fields.add(GraphQLResponseField("__typename").alias(alias))
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as ContributionCalendarMonthResponseProjection
        return Objects.equals(fields, that.fields)
    }

    override fun hashCode(): Int = Objects.hash(fields)

}
